[WGSL] Re-land the command line tool to run the compiler

tadeuzagallo opened the pull request:
#### ed3e8618c480bac24818e081b0e365954f09bf5c
<pre>
[WGSL] Re-land the command line tool to run the compiler
<a href="https://bugs.webkit.org/show_bug.cgi?id=253892">https://bugs.webkit.org/show_bug.cgi?id=253892</a>
rdar://106708690

Reviewed by Nobody (OOPS).

This was originally landed in <a href="https://commits.webkit.org/261911@main">https://commits.webkit.org/261911@main</a>, but later
reverted because it caused build problems. I believed the build issues were caused
by the WebGPU project lacking the `buildImplicitDependencies` flag, and local
testing suggests it fixes the issue.

Original commit message:
Add a `wgslc` command line executable to invoke the compiler. This will be used to
add compiler tests without requiring the rest of WebGPU.

* Source/WebGPU/Configurations/wgslc.xcconfig: Added.
* Source/WebGPU/WGSL/wgslc.cpp: Added.
(printUsageStatement):
(CommandLine::CommandLine):
(CommandLine::file const):
(CommandLine::entrypoint const):
(CommandLine::dumpASTAfterCheck const):
(CommandLine::dumpASTAtEnd const):
(CommandLine::dumpGeneratedCode const):
(CommandLine::parseArguments):
(runWGSL):
(main):
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU.xcodeproj/xcshareddata/xcschemes/WebGPU.xcscheme:

Canonical link: <a href="https://commits.webkit.org/262324@main">https://commits.webkit.org/262324@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ef187fc4fb88ac0616a8b94dc0be868a265e8e86

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1235 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1269 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1308 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2014 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1104 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1321 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1337 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/1266 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1244 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1158 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1149 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1878 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/1182 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1139 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1119 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1129 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1170 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2230 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1178 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1114 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1149 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/308 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1204 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

litherum reviewed (approved):
I don't quite understand how a new target, which should never be built in an automated way, caused a build breakage :/

tadeuzagallo committed:
[WGSL] Re-land the command line tool to run the compiler
https://bugs.webkit.org/show_bug.cgi?id=253892
rdar://106708690

Reviewed by Nobody (OOPS).

This was originally landed in https://commits.webkit.org/261911@main, but later
reverted because it caused build problems. I believed the build issues were caused
by the WebGPU project lacking the `buildImplicitDependencies` flag, and local
testing suggests it fixes the issue.

Original commit message:
Add a `wgslc` command line executable to invoke the compiler. This will be used to
add compiler tests without requiring the rest of WebGPU.

* Source/WebGPU/Configurations/wgslc.xcconfig: Added.
* Source/WebGPU/WGSL/wgslc.cpp: Added.
(printUsageStatement):
(CommandLine::CommandLine):
(CommandLine::file const):
(CommandLine::entrypoint const):
(CommandLine::dumpASTAfterCheck const):
(CommandLine::dumpASTAtEnd const):
(CommandLine::dumpGeneratedCode const):
(CommandLine::parseArguments):
(runWGSL):
(main):
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU.xcodeproj/xcshareddata/xcschemes/WebGPU.xcscheme:

Canonical link: https://commits.webkit.org/262324@main
